test/base/kernel_launch_generic.cpp

```diff
@@ -364,6 +365,33 @@ void run1d_reduction(std::shared_ptr<gko::EXEC_TYPE> exec)
 TEST_F(KernelLaunch, Reduction1D) { run1d_reduction(exec); }
 
 
+void run1d_reduction_cached(std::shared_ptr<gko::EXEC_TYPE> exec,
+                            std::vector<size_type> sizes)
+{
+    gko::array<int64> output{exec, 1};
+    gko::array<char> temp(exec);
+    for (const auto& size : sizes) {
+        temp.clear();
+        gko::kernels::EXEC_NAMESPACE::run_kernel_reduction_cached(
+            exec, [] GKO_KERNEL(auto i) { return i + 1; },
+            [] GKO_KERNEL(auto i, auto j) { return std::max(i, j); },
+            [] GKO_KERNEL(auto j) { return j; }, int64{}, output.get_data(),
+            size, temp);
+
+        ASSERT_EQ(exec->copy_val_to_host(output.get_const_data()),
+                  static_cast<int64>(size));
+        // The temporary storage (used for partial sums) must be smaller than
+        // the input array
+        ASSERT_LE(temp.get_num_elems() / sizeof(int64), size);
```

Comment:
why not the other way round? seems more straightforward to me
```suggestion
        ASSERT_LE(temp.get_num_elems(), size * sizeof(int64));
```